### DISCUSSION TITLE
Manually set locale for Spell Checking Language Feature.

### DISCUSSION THREAD

**Discussion by Atlaimond** [ankurk91/google-chat-electron#117] [Ideas]
System: **Linux Mint 20.03 Xfce** 
App version: **2.19.0**

Hello. 👋️ 
I could use a feature to be able to switch in between spell checker dictionaries. 
Or choose multiple dictionaries similar as [Firefox] does. 
![Screenshot_2022-10-04_03-42-26_crop](https://user-images.githubusercontent.com/75715297/193716713-a2cf4d21-0b6d-4638-85fe-37a9257fcad6.png)

PS. I typo a lot unfortunately, and I'm not native English, so I sometimes do chat with fellow Hungarian people. 🤷‍♂️️ 
Thank you for reading this post in advance! 🙇‍♂️️

**Comment by ankurk91** [comment]
I think electron will detect the locale from your computer and will load that language for spell check automatically

**Reply by Atlaimond** [reply]
> I think electron will detect the locale from your computer and will load that language for spell check automatically

I have done a chat from a few hours ago with a member of my family. We had chatted almost half an hour, exchanged many messages and all the time my words were underlined red, in the chat box, and as I left clicked on them it suggested English words to me to replace. 
 
I know what you're talking about. [Discord] _(in same Linux Mint)_ does this exactly like you described. After, 3-4 words it automatically switches to Hungarian and the red underlines disappear what was marked before as typos. 
 
So, is [Google-Chat-Electron] supposed to this right now too? 😮️

**Comment by ankurk91** [comment]
I will look into it, can't promise.
It requires me to setup a VM with a different language than English.

**Reply by Atlaimond** [reply]
> I will look into it, can't promise. It requires me to setup a VM with a different language than English.

Thank you, but I think I was being misunderstood. It's my fault, sorry for that. 😞️ My entire desktop system runs in English. In [Firefox] and [Thunderbird] I've just set up different dictionaries, alongside English, for spell-checking. In [Discord] case, I've mentioned before, it does this without anything to do for it. 👍️ 
 
So, if there would be possible [Google-Chat-Electron] remain in English just use a different, switchable, dictionary for spell-checking, that is what do the best, if anyone asks me. 😇️ 👍️

**Comment by ankurk91** [comment]
You can read this doc
https://www.electronjs.org/docs/latest/tutorial/spellchecker

```
By default the spellchecker will enable the language matching the current OS locale.
```

**Reply by Atlaimond** [reply]
I see. I'm going to give a shot in this week. Since I guess it requires some digging into to do it. 
Could you point me where is that file located which needs to be edited with those code to make this happen? I hope I will be able to figure out the rest. 🙏️

**Reply by ankurk91** [reply]
Do you want to edit the source code of this app?

**Reply by Atlaimond** [reply]
Not exactly. I want to make work the Hungarian spellchecker while keep the English spellchecker around in [Google-Chat-Electron]. 
As far as I see, I have not much of a choice here, so I have to give it a shot. 🤷‍♂️️ 
 
So far, the most intricate thing I have done on computers when I made [TEoS V: Skyrim] to use the Numeric Keys on my keyboard, since I'm left handed, and I use the arrow keys to move, and the numeric keys are closer to my right hand then the numbers above the letters. 
I had to figure out what file I need to edit, and how to know what. I found the file and figured out it uses [DIK KeyCodes]. I found the codes on Wikipedia, then I was on the right track and after three days, I fortunately made it work correctly! 😋️👍️ 
 
Now, in [Google-Chat-Electron] case, I tried to figure out where is the file, but so far, I feel I'm clueless at the moment. 🤷‍♂️️😞️ 
Maybe today, I might make any progress! I'll try open one by one the files which may contain codes similar looking what the tutorial, you suggested here, contains. 
In the worst case I'll make a huge pile of mess. 🤷‍♂️️ We will see. 😶️

**Reply by ankurk91** [reply]
The link I shared was from electron js docs.

Electron allows to change the spell checker locale despite of your system locale.


It means this app can use a different spell checker locale and your os/system can have a different.

It requires to write code (the line specified on that link).

Unfortunately, I don't have time to make spell check locale configurable.

I don't think you will be able modify a file on your os to make it work

**Reply by Atlaimond** [reply]
Yay! I made it! 😃️ I still cannot believe it took me less then a day! 🥲 I was lucky I guess. 😇️ 

**/home/ _(username)_ /.config/google-chat-electron/Preferences**
and there's only one line in it 
`{"electron":{"media":{"device_id_salt":"7B09D3B5D2EC483EE3EBA3662E5CB4E5"}},"spellcheck":{"dictionaries":["en-GB"],"dictionary":""}}` 
I used my little common sense and edited it to 
`{"electron":{"media":{"device_id_salt":"7B09D3B5D2EC483EE3EBA3662E5CB4E5"}},"spellcheck":{"dictionaries":["hu-HU"],"dictionary":""}}`
and it works! 🤩️ 👍️ 
![Screenshot_2022-10-05_04-56-18](https://user-images.githubusercontent.com/75715297/193972077-35499f69-61e4-4e5b-8c8b-6e20115859f4.png)
 
Although, I have to do this again-and-again every time before I chat with someone non-English, but it is more than nothing. 🤷‍♂️️ I think I'll make just a link to that file to the Desktop, so I can open it with a text editor fast, and do this quick change. 🤔️ I will see to it. 🙂️👍️ 
Maybe one day you'll have the time to implement something to put this into a more comfortable way than this mess I made here. 😅️ Until then, I have this "B" plan, I accidentally discovered, to apply. ☝️🤓️💡️ 
 
Thank you for replying. This is a very good and useful application already you made. 🙂️👌️ It'll be just better in time. 😉️👍️

**Comment by ankurk91** [comment]
I think you can specify multiple locale in that file.

**Reply by Atlaimond** [reply]
> I think you can specify multiple locale in that file.

You're super right! :nerd_face: :+1: 
I googled the `{"spellcheck":{"dictionaries":[` part of the code, in hope Google would come up with something my common sense can do anything, and both DID! :smiley: 

I found someone commented part of the code I recognised the pattern mostly, 
`["en-US", "fr-FR"]`
_Source: https://github.com/nwjs/nw.js/issues/4954#issuecomment-606125429_ 
so I just took how [rowenbuzz](https://github.com/rowenbuzz) separates, by a comma and space, two language-like parts of the code and I just imitated this pattern adapted it to my needs! :pray: :innocent: 
And it has worked again! :star_struck: Yay! :raised_hands: :smile:  It works just like as the [Firefox] multiple-and-simultaneous spellchecking works! :grinning::grey_exclamation: 

![Screenshot_2022-10-05_12-38-50](https://user-images.githubusercontent.com/75715297/194045661-c07a8d1c-cf3e-4dbd-bcda-96910c8903b1.png)

It also suggests typo-replacements from both languages too now! :smiley: 
![Screenshot_2022-10-05_12-39-28](https://user-images.githubusercontent.com/75715297/194045687-47747a0b-f49f-4ee0-b0f2-fa2a5e6c0e1c.png)
went = English :gb: | teremt = Hungarian :hungary: 

**PS.** One thing also caught the attention of my common sense is that; Based on a comment, it looks like not every language code follows the same pattern of `non-capital` `dash` `same-but-capital` pattern. :thinking: 
Like Spanish `"es"` uses a code with no dash and only non-capital letters. . . For some reason. :man_shrugging:⁉️
_Source: https://github.com/nwjs/nw.js/issues/4954#issuecomment-604766136_ 
Just for those to know who end up here looking for solution. :man_shrugging: 

Thank you for the great idea! Now I can use [Google-Chat-Electron] without being disturbed by doing any extra for this feature! 🎆️:star_struck:🎆️ 
This is great! :star2: :man_dancing: :star2: _(victory music :musical_note:)_
